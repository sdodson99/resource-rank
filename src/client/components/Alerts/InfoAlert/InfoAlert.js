import React from 'react';
import PropTypes from 'prop-types';
import styles from './InfoAlert.module.css';
import Alert from '../Alert/Alert';
import { Info } from '@material-ui/icons';

const InfoAlert = ({ children, border, scrollTo }) => (
  <div className={styles.InfoAlert} data-testid="InfoAlert">
    <Alert
      border={border}
      scrollTo={scrollTo}
      className="text-blue-800 border-blue-200 bg-blue-50"
      icon={<Info />}
    >
      {children}
    </Alert>
  </div>
);

InfoAlert.propTypes = {
  children: PropTypes.node,
  border: PropTypes.bool,
  scrollTo: PropTypes.bool,
};

InfoAlert.defaultProps = {};

export default InfoAlert;
