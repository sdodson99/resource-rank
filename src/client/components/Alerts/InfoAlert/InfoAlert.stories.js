import React from 'react';
import InfoAlert from './InfoAlert';

export default {
  component: InfoAlert,
  title: 'Components/Alerts/InfoAlert',
};

const Template = (args) => <InfoAlert {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: 'Successfully updated information.',
};

export const Border = Template.bind({});
Border.args = {
  ...Default.args,
  border: true,
};
