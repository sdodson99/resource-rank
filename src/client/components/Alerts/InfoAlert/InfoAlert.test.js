import React, { createRef } from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { createRenderer } from 'react-test-renderer/shallow';
import InfoAlert from './InfoAlert';
import { useIntersectionObserver } from 'react-intersection-observer-hook';

jest.mock('react-intersection-observer-hook');
useIntersectionObserver.mockReturnValue([createRef(), {}]);

describe('<InfoAlert />', () => {
  it('should mount', () => {
    render(<InfoAlert />);

    const infoAlert = screen.getByTestId('InfoAlert');

    expect(infoAlert).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const component = createRenderer().render(<InfoAlert />);

    expect(component).toMatchSnapshot();
  });
});
