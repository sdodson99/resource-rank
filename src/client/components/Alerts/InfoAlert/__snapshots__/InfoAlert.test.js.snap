// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoAlert /> should render correctly 1`] = `
<div
  className="InfoAlert"
  data-testid="InfoAlert"
>
  <Alert
    className="text-blue-800 border-blue-200 bg-blue-50"
    icon={<Memo />}
  />
</div>
`;
